libstuff/STime.cpp

```diff
@@ -55,11 +55,15 @@ string SCURRENT_TIMESTAMP_MS() {
     return timestamp + "." + msString;
 }
 
-string SFirstOfNextMonth(const string& timeStamp) {
+string SFirstOfMonth(const string& timeStamp, const int64_t& offset) {
+
+    if (offset < 0) {
+        STHROW("500 Offset may only be positive");
```

Comment:
Why? Let's make this general purpose and let it work for any month, including last month or whenever.

In fact, let's call it `SFirstOfMonth` and pass `timestamp` and `monthsInFuture`. If you pass `0`, you get the first of this month, if you pass `-1` you get last month. If you pass `1` you get next month, if you pass `12` you get this month next year, etc. Set the default for `monthsInFuture` to 0 or 1, your choice.

Reply:
Agree with tyler -- we should just make this a general date manipulation function for finding the first of any month in relation to the current timestamp